### PR TITLE
KAFKA-2573: Mirror maker system test hangs and eventually fails

### PR DESCRIPTION
**Pull request by SinghAsDev**

### CODE REVIEW

**Comment by SinghAsDev** [comment]
@guozhangwang @gwenshap mind taking a look? Thanks!

**Comment by SinghAsDev** [comment]
testAclManagementAPIs has been failing on all PRs, not related to changes in this PR.

File: tests/kafkatest/sanity_checks/test_mirror_maker.py [hunk 1]
```diff
@@ -78,13 +99,28 @@ def test_end_to_end(self):
...
         self.consumer.start()
-        self.consumer.wait()
+        node = self.consumer.nodes[0]
 
-        num_consumed = len(self.consumer.messages_consumed[1])
+        wait_until(lambda: self.consumer.alive(node),
+            timeout_sec=10, backoff_sec=.2, err_msg="Consumer was too slow to start")
+        self.logger.info("consumer started in %s seconds " % str(time.time() - t0))
+
+        # Verify that log output is happening
```
Comment (granders) [inline tests/kafkatest/sanity_checks/test_mirror_maker.py]: Delete this block - this was a specific check for ensuring log output in the test_console_consumer

**Comment by granders** [comment]
@SinghAsDev 
It's probably best to move the `alive` check into `ConsoleConsumer.start_node` to make this type of failure easier to immediately detect and pinpoint. 

```
def start_node(self, node):
    super(ConsoleConsumer, self).start_node(node)
    wait_until(lambda: self.alive(node), timeout_sec=10, backoff_sec=.2,
                   err_msg="Timed out waiting for consumer to start.")
```

Also, note that `BackgroundThreadService.wait` has a default timeout of 10 minutes, to prevent it from hanging forever:

```
    def wait(self, timeout_sec=600):
        """Wait no more than timeout_sec for all worker threads to finish.

        raise TimeoutException if all worker threads do not finish within timeout_sec
        """
        ...
```

**Comment by granders** [comment]
@guozhangwang @gwenshap 
This looks good to me, and all system tests all in his branch:
http://jenkins.confluent.io/job/kafka_system_tests_branch_builder/81/console

**Comment by guozhangwang** [comment]
LGTM, thanks.

**Pull request closed without merging**